pkg/domain/ecosystem/EcosystemState.go

```diff
@@ -0,0 +1,19 @@
+package ecosystem
+
+import (
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"golang.org/x/exp/maps"
+)
+
+type EcosystemState struct {
+	InstalledDogus               map[common.SimpleDoguName]*DoguInstallation
+	InstalledComponents          map[common.SimpleComponentName]*ComponentInstallation
+	GlobalConfig                 map[common.GlobalConfigKey]*GlobalConfigEntry
+	DoguConfig                   map[common.DoguConfigKey]*DoguConfigEntry
+	EncryptedDoguConfig          map[common.SensitiveDoguConfigKey]*SensitiveDoguConfigEntry
+	DecryptedSensitiveDoguConfig map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue
+}
+
+func (state EcosystemState) GetInstalledDoguNames() []common.SimpleDoguName {
+	return maps.Keys(state.InstalledDogus)
```

Comment:
It is not hard to write a generic util function for this. With that we can get rid of this dependency.

Reply:
Our current strategy for experimental features like this in the blueprint-operator was, that we can use them if they are very easy to implement by ourselves. We can change this decision. If we do that, we should also remove other usages of experimental functions (e.g. `maps.Values`)